src/common/buffer.cc

```diff
@@ -350,7 +351,7 @@ static ceph::spinlock debug_lock;
   }
   ceph::unique_leakable_ptr<buffer::raw> buffer::create_small_page_aligned(unsigned len) {
     if (len < CEPH_PAGE_SIZE) {
-      return create_aligned(len, CEPH_BUFFER_ALLOC_UNIT);
+      return create_aligned(len, CEPH_BUFFER_ALLOC_UNIT_BASE);
```

Comment:
`buffer::create_small_page_aligned()` is used also by `KernelDevice` when calling a `readv` variant. Not 100% sure we can change the alignment here. However, passing `CEPH_BUFFER_ALLOC_UNIT` before might be actually a misbehavior-without-a-consequence when `CEPH_PAGE_SIZE == CEPH_BUFFER_ALLOC_UNIT`.

Reply:
hrm, if this is the case it would be a bug in KernelDevice I think?  We should change it to use CEPH_PAGE_SIZE in that case, but I don't know if it matters?

Reply:
I don't know too, to be honest. There are many users and verification could be painful. What worries me is we're introducing pretty significant change in behavior potentially resulting in a performance regression that could very hard to find latter.

IIRC currently on all environments, where `PAGE_SIZE` is 4 KB, `create_small_page_aligned()` behaves exactly like `create_page_aligned()`. The difference becomes pretty visible on exotic ones according the description of cc59da9785730a4247a24f2af17401124c506293. Maybe we should always pass 4 KB here.

Reply:
O_DIRECT requires aligning to the disk block size - we should be 4k aligned to work with 4k drives in KernelDevice. https://people.redhat.com/msnitzer/docs/io-limits.txt - if we change the alignment here we need to audit all callers